spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/MultipleRetryPolicy.scala

```diff
@@ -5,29 +5,56 @@ import com.datastax.driver.core.policies.RetryPolicy.RetryDecision
 import com.datastax.driver.core.{ConsistencyLevel, Statement, WriteType}
 
 /** Always retries with the same CL, constant number of times, regardless of circumstances */
-class MultipleRetryPolicy(maxRetryCount: Int, retryDelay: CassandraConnectorConf.RetryDelayConf)
+class MultipleRetryPolicy(maxRetryCount: Int)
   extends RetryPolicy {
 
-  private def retryOrThrow(cl: ConsistencyLevel, nbRetry: Int): RetryDecision = {
+  private def retryManyTimesOrThrow(cl: ConsistencyLevel, nbRetry: Int): RetryDecision = {
     if (nbRetry < maxRetryCount) {
-      if (nbRetry > 0) {
-        val delay = retryDelay.forRetry(nbRetry).toMillis
-        if (delay > 0) Thread.sleep(delay)
-      }
       RetryDecision.retry(cl)
     } else {
       RetryDecision.rethrow()
     }
   }
 
-  override def onReadTimeout(stmt: Statement, cl: ConsistencyLevel,
-                             requiredResponses: Int, receivedResponses: Int,
-                             dataRetrieved: Boolean, nbRetry: Int) = retryOrThrow(cl, nbRetry)
+  private def retryOnceOrThrow(cl: ConsistencyLevel, nbRetry: Int): RetryDecision = {
+    if (nbRetry == 0) {
+      RetryDecision.retry(cl)
+    } else {
+      RetryDecision.rethrow()
+    }
+  }
 
-  override def onUnavailable(stmt: Statement, cl: ConsistencyLevel,
-                             requiredReplica: Int, aliveReplica: Int, nbRetry: Int) = retryOrThrow(cl, nbRetry)
+  override def onReadTimeout(
+      stmt: Statement,
+      cl: ConsistencyLevel,
+      requiredResponses: Int,
+      receivedResponses: Int,
+      dataRetrieved: Boolean,
+      nbRetry: Int): RetryDecision = {
 
-  override def onWriteTimeout(stmt: Statement, cl: ConsistencyLevel, writeType: WriteType,
-                              requiredAcks: Int, receivedAcks: Int, nbRetry: Int) = retryOrThrow(cl, nbRetry)
+    retryManyTimesOrThrow(cl, nbRetry)
+  }
 
+  override def onWriteTimeout(
+      stmt: Statement,
+      cl: ConsistencyLevel,
+      writeType: WriteType,
+      requiredAcks: Int,
+      receivedAcks: Int,
+      nbRetry: Int): RetryDecision = {
+
+    retryManyTimesOrThrow(cl, nbRetry)
+  }
+
+  override def onUnavailable(
+      stmt: Statement,
+      cl: ConsistencyLevel,
+      requiredReplica: Int,
+      aliveReplica: Int,
+      nbRetry: Int): RetryDecision = {
+
+    // We retry once in hope we connect to another
+    // coordinator that can see more nodes (e.g. on another side of the network partition):
+    retryOnceOrThrow(cl, nbRetry)
```

Comment:
But there is one retry here.
There is no point in retrying more often, because on unavailable, there is no timeout to slow down retrying.